dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -91,85 +91,207 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
 
   def listPendingEventDbs(): Future[Vector[EventDb]] = eventDAO.getPendingEvents
 
-  def listEvents(): Future[Vector[Event]] = {
-    for {
-      rValDbs <- rValueDAO.findAll()
-      eventDbs <- eventDAO.findAll()
-      outcomes <- eventOutcomeDAO.findAll()
-    } yield {
-      val rValDbsByNonce = rValDbs.groupBy(_.nonce)
-      val outcomesByNonce = outcomes.groupBy(_.nonce)
-      eventDbs.map(db =>
-        Event(rValDbsByNonce(db.nonce).head, db, outcomesByNonce(db.nonce)))
+  def listEvents(): Future[Vector[OracleEvent]] = {
+    eventDAO.findAll().map { eventDbs =>
+      val events = eventDbs.groupBy(_.eventDescriptorTLV)
```

Comment:
Why do we need the `.groupBy()` here? Is there some sort of implicit sorting required?

Reply:
I am using `groupBy` here so we can get all the eventDbs with the same `eventDescriptorTLV`, ones with the same `eventDescriptorTLV` will be for the same event, ie signing a large number.